File: frontend/frontend-administration/src/app/models/ProfesseurModel/professeur.ts
import {Departement} from "../enums/departement";


export class Professeur {
  private id!: number;
  private nom!: string;
  private prenom!: string;
  private email!: string;
  private password!: string;
  private etablissement!: string;
  private departement!: Departement;
  private discipline!: Departement[];
}

File: backend/reservationSalle-service/src/main/java/fsm/miaad/models/Professeur.java
package fsm.miaad.models;

import fsm.miaad.entities.Salle;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.Id;

import java.util.List;


@Data @AllArgsConstructor @NoArgsConstructor @ToString @Builder
public class Professeur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom,prenom;
    @Enumerated(EnumType.STRING)
    private Departement departement;


}

